app/javascript/hyacinth_ui_v1/components/digital_objects/new/AssetNew.js

```diff
@@ -41,11 +40,14 @@ function AssetNew(props) {
   };
   return (
     <Form>
-      <input id={elementId} type="file" required onChange={onChange} parentid={parentId} direct_upload="true" />
+      <input id={AssetNew.elementId} type="file" required onChange={onChange} parentid={parentId} direct_upload="true" />
     </Form>
   );
 }
 
+AssetNew.elementId = `input-${String.toString(Math.random()).replace('.', '')}`;
```

Comment:
I think this goes against good reusable component design, since you're effectively creating a static variable that will be the same for every occurrence of this component for the lifetime of the react app.  Things like unique element id would be better off set in the component's state.